eng/build.ps1

```diff
@@ -59,38 +59,32 @@ if ($help -or (($null -ne $properties) -and ($properties.Contains('/help') -or $
   exit 0
 }
 
-# VS Test Explorer support for libraries
 if ($vs) {
-  Write-Host "VS Test Explorer now works without needing to call build.cmd. The -vs switch will be removed eventually."
   . $PSScriptRoot\common\tools.ps1
 
-  # Microsoft.DotNet.CoreSetup.sln is special - hosting tests are currently meant to run on the
-  # bootstrapped .NET Core, not on the live-built runtime.
-  if (([System.IO.Path]::GetFileName($vs) -ieq "Microsoft.DotNet.CoreSetup.sln") -or ($vs -ieq "Microsoft.DotNet.CoreSetup")) {
+  if (-Not (Test-Path $vs)) {
+    $solution = $vs
+    # Search for the solution in libraries
+    $vs = Join-Path (Join-Path (Split-Path $PSScriptRoot -Parent) "\src\libraries") $vs | Join-Path -ChildPath "$vs.sln"
```

Comment:
Nit: I would stick to either parentheses or pipes, but I would not mix those two styles.  Either:
```ps1
Join-Path (Join-Path (Join-Path (Split-Path $x -Parent) "src\libraries") $vs) "$vs.sln"
```
or
```ps1
Split-Path $x -Parent | Join-Path -ChildPath "src\libraries" | Join-Path -ChildPath $vs | Join-Path -ChildPath "$vs.sln"
```
(The backslash before `src` may be omitted.)